modules/resolve.js

```diff
@@ -0,0 +1,83 @@
+export const resolveLocation = (location, base) => {
+  if (typeof location === 'string') {
+    return resolve(location, base)
+  } else {
+    location.pathname = resolve(location.pathname, base)
```

Comment:
Adding a trailing slash should work, not sure why I didn't think of that before.

I think that I will need to add a check so that resolving `..` with `/a/b/c` returns `/a/b` and not `/a/b/`, but otherwise no issues with that comes to mind.